protocols/bittorrent.ts

```diff
@@ -0,0 +1,94 @@
+import { Static } from '@sinclair/typebox'
+import Protocol, { Ctx, SyncOptions } from './interfaces'
+import { BitTorrentProtocolFields } from '../api/schemas'
+
+import path from 'node:path'
+import { cp } from 'node:fs/promises'
+
+import { TorrentManager } from 'bt-fetch'
+
+export interface BitTorrentProtocolOptions {
+  path: string
+}
+
+export class BitTorrentProtocol implements Protocol<Static<typeof BitTorrentProtocolFields>> {
+  options: BitTorrentProtocolOptions
+  manager: TorrentManager | null
+
+  constructor (options: BitTorrentProtocolOptions) {
+    this.options = options
+    this.manager = null
+  }
+
+  async load (): Promise<void> {
+    const folder = this.options.path
+    this.manager = new TorrentManager({ folder })
+  }
+
+  async unload (): Promise<void> {
+    await this.manager?.destroy()
+  }
+
+  async sync (id: string, folderPath: string, options?: SyncOptions, ctx?: Ctx): Promise<Static<typeof BitTorrentProtocolFields>> {
+    ctx?.logger.info('[bittorrent] Sync Start')
+
+    const link = `bittorent://${id}/`
+
+    if (this.manager === null) {
+      throw new Error('[bittorrent] Torrent Manager Not Initialized')
+    }
+
+    const manager: TorrentManager = this.manager
+
+    // Create keypair from the site ID and the seed key
+    const { publicKey, secretKey } = manager.createKeypair(id)
+
+    const storageFolder = path.join(this.options.path, 'data', publicKey, id)
```

Comment:
do we create a new storage folder per `publicKey`, `id` pair? Shouldn't there only be one `publicKey` per `id`?

Reply:
ensure that `api/index.ts` we instantiate `ConcreteProtocolManager` with the right path

```ts
  const protocols = new ConcreteProtocolManager({
    ipfs: {
      path: path.join(protocolStoragePath, 'ipfs'),
      provider: cfg.ipfsProvider
    },
    hyper: {
      path: path.join(protocolStoragePath, 'hyper')
    },
    http: {
      path: path.join(protocolStoragePath, 'http')
    },
    bittorrent: {
      path: path.join(protocolStoragePath, 'bittorrent')
    }
  })
```

Reply:
The `publicKey/id` thing is needed to account for how WebTorrent stores torrents. It requires torrent data to be stored in a folder with the given `name` for the torrent. It's a bit wonky but fighting it will actually be way more effort. 😅